ui/shared/TruncatedText.jsx

```diff
@@ -22,8 +22,10 @@ export default class TruncatedText extends React.Component {
         </p>
         {text.length > maxLength && (
           <Button
-            color="link"
-            className={`${showMoreClass} d-block p-0 ml-auto`}
+            color={color || 'link'}
```

Comment:
You'll need to add `color` as a prop type below, and instead of doing this logic you could set 'link' as a default prop. However, I think setting it to "secondary" would be best since we only use the link color for actual links.

Reply:
Edit: I just realized that the 'link' class is what's making it look bolded without the outline, correct? So I think you can just set that to the default prop.

Reply:
Actually there are a couple of verification that are performed, in `outline` and `className`:
If it has the button style:
```jsx
color="info" // for example
outline={true} // set outlined button
className="d-block ml-auto" // aligns to the right
```

If it has the link style:
```jsx
color="link" // always
outline={false}
className="font-weight-bold text-reset d-block ml-auto" // set font as bold, inherit color, aligns to the right
```

I will paste the snippet of the code I have changed so you can share yours thoughts.

Reply:
Apart from inserting the `color` in prop type, I had to change the a little bit, because `color` would never be `undefined` (or a false value) again.
The `isButtonStyle` was created, because it appeared twice. Although there might be a better way to do this.

```jsx
render() {
    const { text, maxLength, title, color } = this.props;
    const { showMoreResults } = this.state;
    const isButtonStyle = color !== 'link';

    return (
      // ...
          <Button
            color={color}
            outline={isButtonStyle}
            className={`${isButtonStyle ||
              'font-weight-bold text-reset'} d-block ml-auto`}
            // ...
          >
            // ...
          </Button>
      // ...
    );
  }
```

Reply:
Instead of doing isButtonStyle you can define a default prop with 'link' at the bottom of the file: https://reactjs.org/docs/typechecking-with-proptypes.html

You will need to add a color prop type regardless.

Reply:
Ok, I tried it again! Please, don't give up on me.

Reply:
Don't be silly - it's all part of the learning process :)